### DISCUSSION TITLE
How to link Lua statically, when both .a and .so exist in search path configured by pkg-config?

### DISCUSSION THREAD

**Discussion by zopsicle** [mlua-rs/mlua#379] [Q&A]
On my system, both `liblua.a` and `liblua.so` exist in the directory pointed to by the `lua.pc` file used by pkg-config. Currently, `liblua.so` is linked with. Is there a way to link with `liblua.a` instead of `liblua.so`? I do not want to use vendored Lua.

**Comment by khvzak** [comment]
If you don't want vendored build, you always can use custom linking options mentioned in the [Compiling](https://github.com/mlua-rs/mlua?tab=readme-ov-file#compiling) section (setting `LUA_LINK=static` and providing `LUA_LIB`, etc).
